compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/MethodSignatureMapper.kt

```diff
@@ -127,15 +127,19 @@ class MethodSignatureMapper(private val context: JvmBackendContext) {
                 (DescriptorVisibilities.isPrivate(suspendFunctionOriginal().visibility) ||
                         originalForDefaultAdapter?.isInvisibleInMultifilePart() == true)
 
-    private fun IrSimpleFunction.shouldMangleAsInternal(): Boolean =
-        (origin != JvmLoweredDeclarationOrigin.STATIC_INLINE_CLASS_CONSTRUCTOR &&
-                visibility == DescriptorVisibilities.INTERNAL &&
-                !isPublishedApi())
-                || originalForDefaultAdapter?.shouldMangleAsInternal() == true
+    private fun IrSimpleFunction.shouldMangleAsInternal(): IrSimpleFunction? {
```

Comment:
Maybe rename it to something like `getBaseNameForMangling` (`...IfNeeded`?), since it no longer returns boolean?

Reply:
Good idea. I went for: `getInternalFunctionForManglingIfNeeded`.